enhancements/etcd/automated-backups.md

```diff
@@ -0,0 +1,284 @@
+    // +kubebuilder:default:=""
+    // +optional
+    Reason string `json:"reason"`
+    
+    // Schedule defines the recurring backup schedule in Cron format
+      // every 2 hours: 0 */2 * * *
+      // every day at 3am: 0 3 * * *
+    // Setting to an empty string "" means disabling scheduled backups
+    // Default: ""
+    // TODO: Define how we do validation for the format (e.g validation webhooks)
+    // and the limits on the frequency to disallow unrealistic schedules (e.g */2 * * * * every 2 mins)
+    // TODO: What is the behavior if the backup doesn't complete in the specified interval
+        // e.g: every 1hr but the backup takes 2hrs to complete
+        // Wait for the current one to complete?
+        // And do we generate an alert even if MaxDurationToCompleteBackup is not set, since the scheduled interval has passed?
+    Schedule string `json:"schedule"`
+    
+    // MaxDurationToCompleteBackup sets the max duration after which if
+    // an initiated backup has not successfully completed a critical alert will be generated.
+    // The format is as accepted by time.ParseDuration(), e.g "1h10m"
+    // See https://pkg.go.dev/time#ParseDuration
+    // Setting to "0" disables the alert.
+    // Default: "1h"
+    // TODO: Validation on min and max durations.
+        // The max should be smaller than the scheduled interval
+        // for the next backup else the alert will never be generated
+    MaxDurationToCompleteBackup string `json: "maxDurationWithoutBackup"`
+    
+    // RetentionCount defines the maximum number of backups to retain.
+    // If the number of successful backups matches retentionCount 
+    // the oldest backup will be removed before a new backup is initiated.
+    // The count here is for the total number of backups
+    // TODO: Validation on min and max counts.
+    // TODO: Retention could also be based on the total size, or time period
+        // e.g retain 500mb of backups, or discard backups older than a month
+        // If we want to extend this API in the future we should make retention
+        // a type so we can add more optional fields later
+    RetentionCount int `json: "retentionCount"`
+    
+}
+```
+
+**TODO:** How do we feature gate the entire type?
+For a field within a non-gated type it seems to be the tag [`// +openshift:enable:FeatureSets=TechPreviewNoUpgrade`](https://github.com/openshift/api/blob/master/example/v1/types_stable.go#L33). No such tag visible on the [whole type gated example](https://github.com/openshift/api/blob/1957a8d7445bf2332f027f93a24d7573f77a0dc0/example/v1alpha1/types_notstable.go#L19).
+
+
+```Go
+// EtcdBackupStatus represents the status of the EtcdBackup CRD.
+type EtcdBackupStatus struct {
+
+    // Conditions represents the observations of the EtcdBackup's current state.
+    // TODO: Identify different condition types/reasons that will be needed.
+    // +optional
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
This surprises me.  Having status on configuration is not common.  Try a thought experiment modelling, "backup now" requests as a separate resource like https://github.com/openshift/api/blob/master/insights/v1alpha1/types_insights.go#L16 and see how it fairs.  I think it gives you some greater flexibility and you'll be able to order "one at a time" with a "brush your teeth policy" of only the latest is honored and all others get updated as "won't do".